sklearn/utils/_testing.py

```diff
@@ -781,15 +782,18 @@ def _convert_container(
     elif constructor_name == "slice":
         return slice(container[0], container[1])
     elif constructor_name == "sparse_csr":
+        # TODO: when we depend on SciPy>=1.8 return `csr_array` instead and change
+        # the case bellow from `sparse_csr_array` to `sparse_csr_matrix` and make
+        # it return a `csr_matrix`.
         return sp.sparse.csr_matrix(container, dtype=dtype)
     elif constructor_name == "sparse_csr_array":
         if sp_version >= parse_version("1.8"):
-            return sp.sparse.csr_array(container, dtype=dtype)
+            return sp.sparse.csc_array(container, dtype=dtype)
```

Comment:
```suggestion
            return sp.sparse.csr_array(container, dtype=dtype)
```

Reply:
@y-vectorfield you forgot this one.

Reply:
Oh, I'm sorry. I reflected your comment.